File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    const inbox = new Set()
    for (i = 1; i <= elements.length; i++) {
        const newElements = elements.concat(elements.slice(0, i));
         for (k = 0; k < elements.length; k++) {
             inbox.add(newElements.slice(k, k + i).reduce((a, b) => a + b))
         }
    }
    return inbox.size
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    inbox = set()
    n = len(elements)
    for i in range(1, n + 1):
        temp_elements = elements + elements[:i-1] 
        temp_elements = elements + elements[:i]
        for k in range(n):
            current_sum = sum(temp_elements[k : k + i])
            inbox.add(current_sum)
            
    return len(inbox)
